src/JointTrajectoryControllerBase.cpp

```diff
@@ -349,6 +374,25 @@ void JointTrajectoryControllerBase::goalCallback(GoalHandle goalHandle)
   newContext->mTrajectory = trajectory;
   newContext->mGoalHandle = goalHandle;
 
+  // Evaluate the trajectory at the current time.
+  auto offsetDesiredState = mControlledSpace->createState();
+  Eigen::VectorXd offsetDesiredPosition(mControlledSpace->getDimension());
+  trajectory->evaluate(0, offsetDesiredState);
+  mCompoundSpace->logMap(offsetDesiredState, offsetDesiredPosition);
+
+  Eigen::VectorXd offsetActualPosition(mControlledSpace->getDimension());
+  offsetActualPosition = mControlledSkeleton->getPositions();
+
+  Eigen::VectorXd offset(mControlledSpace->getDimension());
+  offset = offsetDesiredPosition - offsetActualPosition;
+  
+  offset = (offset / (2*M_PI));
```

Comment:
Can we add explanatory comments here, e.g. why we're making `offset` be multiple of `2pi`?
I'd also prefer using `auto multiplier = offset/ (2*M_PI)` and do `offset(i) = round(multiplier(i)) * 2pi` for readability. You're mixing the meaning of offset here as well.